autoload/cmake4vim.vim

```diff
@@ -240,25 +240,34 @@ function! cmake4vim#CCMake(...) abort
     " * empty, h - Open ccmake in horizontal split
     " * v - Open ccmake in vertical split
     " * t - Open ccmake in new tab
-    let l:mode = ''
-    if exists('a:1') && a:1 !=# ''
-        let l:mode = a:1
+    let l:mode = a:0 ? a:1 : 'hsplit'
+    let l:supported_modes = split(cmake4vim#CompleteCCMakeModes(0, 0, 0), '\n')
+
+    if index(l:supported_modes, l:mode) == -1
+        call utils#common#Warning('Unsupported window mode: ' . l:mode)
+        return
+    endif
+
+    let l:cmd = 'terminal '
+    if !has('nvim')
+        let l:cmd .= '++close '
     endif
+    let l:cmd .= 'ccmake ' . utils#fs#fnameescape(utils#cmake#getBuildDir())
     if has('nvim')
-        if l:mode ==# 'v'
-            exec 'vsp | terminal ccmake ' utils#fs#fnameescape(utils#cmake#getBuildDir())
-        elseif l:mode ==# 't'
-            exec 'tabnew | terminal ccmake ' utils#fs#fnameescape(utils#cmake#getBuildDir())
+        if l:mode ==# 'vsplit'
```

Comment:
Maybe something like this?

```
if has('nvim')
let l:cmds = { 'vsplit' : 'vsp | ', 'hsplit': 'sp | ', 'tab' : 'tabnew | ' }
else
let l:cmds = { 'vsplit' : 'vertical ', 'hsplit': '', 'tab' : 'tab ' }
endif
execute l:cmds . l:cmd
```

It's less code.

Reply:
Good catch.